crates/xline-client/src/clients/auth.rs

```diff
@@ -222,33 +222,44 @@ impl AuthClient {
     ///         .await?
     ///         .auth_client();
     ///
-    ///     client.user_add(AuthUserAddRequest::new("user1")).await?;
+    ///     client.user_add("user1", "", None).await?;
     ///     Ok(())
     /// }
     ///```
     #[inline]
-    pub async fn user_add(&self, mut request: AuthUserAddRequest) -> Result<AuthUserAddResponse> {
-        if request.inner.name.is_empty() {
+    pub async fn user_add(
+        &self,
+        name: impl Into<String>,
+        password: impl AsRef<str>,
```

Comment:
Should the password be made optional and put into options?

Reply:
if i use `password: Option<impl AsRef<str>>`, then the `user_add` function should pass a `None::<&str>`, because *type annotations needed*. That's a bit complex.
I prefer to use empty str, because in `xlinectl/src/command/user/add.rs`, user must input a password without `--no_password` given. So we'll always get a String.
If you think Option is important to indicate **no** password, maybe we should use `Option<String>`?

Reply:
That makes sense, please add a comment describing that an empty string means no password.